frontend/src/pages/NewTag/Form.jsx

```diff
@@ -0,0 +1,179 @@
+  display: flex;
+  align-items: center;
+  flex-direction: row;
+  width: fit-content;
+  margin-bottom: 10px;
+`;
+
+const StyledInput = styled.input`
+  flex: 0 0 2em;
+  margin-bottom: 5px;
+  font-size: 16px;
+  border-radius: 4px;
+  color: black;
+  border: ${(props) => (props.valid ? '1px solid springgreen' : '1px solid orangered')};
+  border: ${(props) => (props.empty ? '1px solid darkgray' : null)};
+  outline-color: transparent;
+  outline-style: none;
+`;
+
+const Form = () => {
+  // Router
+  const history = useHistory();
+
+  // Redux
+  const dispatch = useDispatch();
+  const userSelector = useSelector((state) => state.user);
+  const { token } = userSelector.user;
+
+  // State
+  const allTags = useGetTags() || [];
+  const [tag, setTag] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [valid, setValid] = useState(false);
+
+  // Update input teext
+  const onChange = (e) => {
+    setTag(e.target.value);
+  };
+
+  // Only allow tags with letters
+  const validateTag = (_tag) => {
+    const re = /^[a-åA-Å\s]*$/;
+    return re.test(_tag);
+  };
+
+  // Check if the tag already exists
+  const checkIfTagExists = (_tag) => {
+    for (const exsistingTag of allTags) {
+      if (exsistingTag.toLowerCase() === _tag.toLowerCase()) {
+        return true;
+      }
+    }
+    return false;
+  };
+
+  // Check if the tag exists and follows the correct pattern and
+  // sets the border color according to tat
+  const checkTag = () => {
+    if (!validateTag(tag) || checkIfTagExists(tag)) {
+      setValid(false);
+    } else {
+      setValid(true);
+    }
+  };
+
+  // Check tag every time it's changed
+  useEffect(() => {
+    checkTag();
+  }, [tag]);
+
+  // Submit tag to backend on submit
+  const onSubmit = async (e) => {
+    e.preventDefault();
+
+    // Start the loading icon
+    setLoading(true);
+
+    // Remove whitespace on ends
+    const trimmedTag = tag.trim();
+
+    // Check if it follows the patterns
+    if (!validateTag(trimmedTag)) {
+      dispatch(alertActions.error('Tags can only contain letters'));
+      return setValid(false);
+    }
+
+    // Check if it exists
+    if (checkIfTagExists(trimmedTag)) {
+      dispatch(alertActions.error('The tag already exists.'));
+      return setValid(false);
+    }
+
+    // Try to push it to the server
+    try {
+      const response = await fetch('/api/Tag', {
+        method: 'PUT',
+        mode: 'cors',
+        cache: 'no-cache',
+        credentials: 'same-origin',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${token}`,
+        },
+        redirect: 'follow',
+        referrerPolicy: 'no-referrer',
+        body: JSON.stringify({
+          name: trimmedTag,
+        }),
+      });
+
+      // Not okey throw error else
+      // set tag to empty string, stop loading, dispatch message and start tiem out for redirect
+      if (!response.ok || response.status !== 200) {
```

Comment:
Is it necessary to check for the exact response code 200? An "OK" response could be something else (e.g. `201 Created` should also be considered an equally valid response). The backend may currently give a 200, but it could change to something like a 201 in time. Unless there is a particular reason that you do both checks explicitly, I'd change it to only check ok.

Reply:
Because specificity is important on status codes, they mean different things. If the backend changes then the frontend should be changed accordingly, and should in reality also change the frontend's functionality.

Reply:
Fair enough